### PR TITLE
Add log streaming

### LINKED ISSUE #11: View streaming traces for a function
The portal displays traces per function (as opposed to per app like in app service)

### PR DESCRIPTION
**Pull request by EricJizbaMSFT**
Builds will fail until the appservice package is published with logstream support: https://github.com/Microsoft/vscode-azuretools/pull/67

Fixes #11

Edit: More fixes for log stream in the shared package here: https://github.com/Microsoft/vscode-azuretools/pull/68

### CODE REVIEW

File: src/commands/logstream/stopStreamingLogs.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AzureTreeDataProvider, IAzureNode } from 'vscode-azureextensionui';
+import { FunctionAppTreeItem } from '../../tree/FunctionAppTreeItem';
+import { ILogStreamTreeItem } from './ILogStreamTreeItem';
+
+export async function stopStreamingLogs(tree: AzureTreeDataProvider, node?: IAzureNode<ILogStreamTreeItem>): Promise<void> {
```
Comment (PrashanthCorp) [inline src/commands/logstream/stopStreamingLogs.ts]: Looks like this function (and corresponding context menu item) would be visible for every Function App. Is there a way to filter the node picker and context menu presence down to Apps that have a logStream present? (I see that being done with the `node.treeItem.logStream` conditional - I think the UX might be cleaner though)

File: src/commands/logstream/startStreamingLogs.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import { localize } from '../../localize';
+import { FunctionAppTreeItem } from '../../tree/FunctionAppTreeItem';
+import { nodeUtils } from '../../utils/nodeUtils';
+import { ILogStreamTreeItem } from './ILogStreamTreeItem';
+
+export async function startStreamingLogs(context: vscode.ExtensionContext, actionHandler: AzureActionHandler, tree: AzureTreeDataProvider, node?: IAzureNode<ILogStreamTreeItem>): Promise<void> {
+    if (!node) {
+        node = <IAzureNode<ILogStreamTreeItem>>await tree.showNodePicker(FunctionAppTreeItem.contextValue);
+    }
+
```
Comment (PrashanthCorp) [inline src/commands/logstream/startStreamingLogs.ts]: I don't see a check to verify that the Streaming Log for the chosen node hasn't already been started. I don't see anything in the docs suggesting that `createOutputChannel` would perform an 'already-present' check.

Comment (PrashanthCorp) [inline src/commands/logstream/startStreamingLogs.ts]: I don't see a check to verify that the Streaming Log for the chosen node hasn't already been started. I don't see anything in the docs suggesting that `createOutputChannel` would perform an 'already-present' check.

File: src/commands/logstream/stopStreamingLogs.ts (hunk 1 above)
Comment (EricJizbaMSFT) [reply src/commands/logstream/stopStreamingLogs.ts]: It's not recommended to hide things like context-menu commands based on state. Users will get confused about why it displays sometimes and not other times. Best practice is actually to gray-out the option, but I don't think that's possible in VS Code

In any case, I'll just display a warning to the user in this case.

File: src/commands/logstream/startStreamingLogs.ts (hunk 2 above)
Comment (EricJizbaMSFT) [reply src/commands/logstream/startStreamingLogs.ts]: Fixed

File: src/commands/logstream/stopStreamingLogs.ts (hunk 1 above)
Comment (PrashanthCorp) [reply src/commands/logstream/stopStreamingLogs.ts]: Yeah, greying out an option makes waay more sense. I stand corrected. :)

File: src/commands/logstream/startStreamingLogs.ts (hunk 2 above)
Comment (PrashanthCorp) [reply src/commands/logstream/startStreamingLogs.ts]: Is this fixed in [this](https://github.com/Microsoft/vscode-azuretools/pull/68) AzureTools PR?

Comment (EricJizbaMSFT) [reply src/commands/logstream/startStreamingLogs.ts]: A combination of that PR and the commit I pushed to this PR [here](https://github.com/Microsoft/vscode-azurefunctions/pull/207/commits/eef0de9f45d8ad6285f19c7df6eafc9d303641bd)

**Pull request merged**